local-notifications/android/src/main/java/com/capacitorjs/plugins/localnotifications/LocalNotification.java

```diff
@@ -247,18 +247,28 @@ public static LocalNotification buildNotificationFromJSObject(JSObject jsonObjec
         return notificationsList;
     }
 
-    public static JSObject buildLocalNotificationPendingList(List<String> ids) {
+    public static JSObject buildLocalNotificationPendingList(List<LocalNotification> notifications) {
         JSObject result = new JSObject();
         JSArray jsArray = new JSArray();
-        for (String id : ids) {
-            JSObject notification = new JSObject();
-            try {
-                int intId = Integer.parseInt(id);
-                notification.put("id", intId);
-            } catch (NumberFormatException ex) {
-                notification.put("id", -1);
+        for (LocalNotification notification : notifications) {
+            JSObject jsNotification = new JSObject();
+            jsNotification.put("id", notification.getId());
+            jsNotification.put("title", notification.getTitle());
+            jsNotification.put("body", notification.getBody());
+            LocalNotificationSchedule schedule = notification.getSchedule();
+            if (schedule != null) {
+                JSObject jsSchedule = new JSObject();
+                jsSchedule.put("at", schedule.getAt());
+                jsSchedule.put("every", schedule.getEvery());
+                jsSchedule.put("count", schedule.getCount());
+                jsSchedule.put("on", schedule.getOn());
+                jsSchedule.put("repeats", schedule.isRepeating());
+                jsNotification.put("schedule", jsSchedule);
             }
-            jsArray.put(notification);
+
+            jsNotification.put("extra", notification.getExtra());
+
+            jsArray.put(jsNotification);
```

Comment:
This is missing some notification information.
There is an easier way of getting the original notification as JSObject, I can't put it as suggested change but should be something like
```
try {
    JSObject jsNotification = new JSObject(notification.getSource());
    jsArray.put(jsNotification);
} catch (JSONException e) {}
```

Reply:
On iOS we do not store the original notification object used to create the actual system notification (maybe its worth doing in a separate PR), so I limited what gets returned in the request to the fields that are most likely to be relevant and common across both iOS and Android:

From https://github.com/ionic-team/capacitor-plugins/issues/193
> Would it be possible to create a new feature that takes an ID and return an object containing details such as **title, body, and schedule** for a given local notif?

Reply:
On Android we can return everything, and it's easier to return everything, but if we don't want to return everything because it's not possible on iOS, the it's probably better to not return LocalNotificationSchema as if we return that, users would expect to get the whole LocalNotificationSchema that was originally sent to schedule method.

Reply:
Good point